main.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+const {context} = require("@arangodb/locals");
+const createRouter = require("@arangodb/foxx/router");
+const {query, db} = require("@arangodb");
+const graph_module = require("@arangodb/general-graph");
+const queues = require("@arangodb/foxx/queues");
+const joi = require("joi");
+const {modelTypes, metadataCollectionName} = require("./scripts/setup");
+
+const router = createRouter();
+context.use(router);
+
+// TODO: Make this model specific?
+const batch_size = 64;
+exports.BATCH_SIZE = batch_size;
+
+const embeddingQueueName = "embeddings_generation";
+
+function retrieveModel(modelName, modelType) {
```

Comment:
I'd move these into separate utility files. Personally I prefer one function per utility file but that's again a matter of taste.